File: mylibrary/src/main/java/bongmi/mylibrary/UserDao.java
package bongmi.mylibrary;

import java.util.List;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

/**
 * Copyright (c) 2018, Bongmi
 * All rights reserved
 * Author: wangjunjie@bongmi.com
 */

@Dao
public interface UserDao {
  @Query("SELECT * FROM user limit 1")
  User get();

  @Query("SELECT * FROM user")
  List<User> getAll();

  @Query("SELECT * FROM user WHERE id IN (:userIds)")
  List<User> loadAllByIds(int[] userIds);

  @Query("SELECT * FROM user WHERE first_name LIKE :first AND "
      + "last_name LIKE :last LIMIT 1")
  User findByName(String first, String last);

  @Insert
  void insertAll(User... users);

  @Delete
  void delete(User user);

  @Query("delete from user")
  void deleteAll();

  @Update
  void update(User user);

  @Query("select count(*) from user")
  int count();
}

File: mylibrary/src/main/java/bongmi/mylibrary/UserDao.kt
package bongmi.mylibrary

import android.arch.persistence.room.*

/**
 * Copyright (c) 2018, Bongmi
 * All rights reserved
 * Author: wangjunjie@bongmi.com
 */

@Dao
interface UserDao {
  @Query("SELECT * FROM user limit 1")
  fun get(): User

  @Query("SELECT * FROM user")
  fun getAll(): List<User>

  @Query("SELECT * FROM user WHERE id IN (:userIds)")
  fun loadAllByIds(userIds: IntArray): List<User>

  @Query("SELECT * FROM user WHERE first_name LIKE :first AND last_name LIKE :last LIMIT 1")
  fun findByName(first: String, last: String): User

  @Query("SELECT * FROM user WHERE :where LIMIT 1")
  fun testFindByName(where: String): User

  @Insert
  fun insertAll(users: Array<User>)

  @Delete
  fun delete(user: User)

  @Query("delete from user")
  fun deleteAll()

  @Update
  fun update(user: User)

  @Query("select count(*) from user")
  fun count(): Int
}
